### DISCUSSION TITLE
JS Web Socket Wrapper Libraries

### DISCUSSION THREAD

**Discussion by evans23** [django/channels#1808] [Q&A] [answered]
Hi,

I'm working on a website with a Vue.js frontend that hits a DRF backend and I'm currently implementing web sockets for our real time updating. I have native web sockets functional in the browser, but I was wondering if there are any current recommended client side wrapper libraries that work well with channels?

I'm new with channels, but from what I can tell, there should be a way to create a reconnecting web socket that falls back on HTTP Long polling similar to socket.io for expressjs. I found the npm django_channels package, but that appears to be explicitly disowned by this project.

Thanks for your help!

**Comment by carltongibson** [comment]
HI @evans23 — good question. I don't know what folks are using out there, but maybe others can reply...

**Comment by carltongibson** [comment, marked as answer]
@destelio mentioned https://github.com/joewalnes/reconnecting-websocket over on #1685

**Reply by evans23** [reply]
Yeah, I ended up using this repo, https://github.com/pladaria/reconnecting-websocket, since it is in npm and has some fresher commits, and then just accepted that HTTP Long polling buys me very little as a fall back. Thanks for keeping up with this!

**Reply by destelio** [reply]
great!
